_posts/2017-08-10-rubyonrails.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title: "Learn Rails and build web apps"
+date: 2017-08-10 00:00:00 +530
+author: Anant Prakash
+category:
+tags:
+- Ruby on Rails
+- Web development
+---
+
+## Hi reader !
+
+Most of you would have learnt/seen HTML & CSS based static web pages.<br>
+Some of you would have also used Javascript to build dynamic web pages.<br>
+But when it comes to building web apps, you need to write/code server-side to bring life
+to these web pages.
+
+That's where [Ruby on Rails](https://en.wikipedia.org/wiki/Ruby_on_Rails) comes in.
+For those of you who have had their hands on web frameworks would know there are
+quite a few to choose from.
+
+> Here's a [list of web frameworks](https://developer.mozilla.org/en-US/docs/Learn/Server-side/First_steps/Web_frameworks#A_few_good_web_frameworks) for your viewing pleasure.
+
+And sometimes it's gets frustrating to choose which one to use. As it's unethical to force you
+to select any of them.<br>
+Here's some good old praise for Ruby on Rails:
+* [Quora](https://www.quora.com/Why-do-so-many-startups-use-Ruby-on-Rails)
+* [Pros and Cons of Ruby on Rails](https://bitzesty.com/2014/03/03/ruby-on-rails-what-it-is-and-why-we-use-it-for-web-applications)
+* [A senior dev blog](https://www.toptal.com/ruby-on-rails/after-two-decades-of-programming-i-use-rails)
+
+### Tl;dr version of above posts
+
+| Pros | Cons |
+| Rapid development | Not as well known |
+| Wide developer support | Some web hosts don't support it |
+| Ruby is easy yet powerful|  |
+
+## Getting started
+
+Did I tell you how easy it was to learn Ruby on Rails from scratch ?
+
+Have a look at [Rails book chapter 1](https://www.railstutorial.org/book/beginning).
+If you follow through the first chapter you would have already built a "Hello World" web app.
+How's that for 'Rapid development' ?
+
+Following the mentioned book would itself make you fluent with Ruby on Rails.
+
+## Some advice on learning Ruby on Rails
+
+* You don't need to be proficient in Ruby to learn Rails. [Rails book chapter 4](https://www.railstutorial.org/book/rails_flavored_ruby) covers most of ruby you'll need. A point to note is that rails in itself has lots of
+functionality so it is better to use them instead of writing Ruby code for them.
+* The chapter on MVC (Model View Controller) forms the basis of Rails web app so its recommended
```

Comment:
I was thinking, is it possible to write a post on MVC and have rails as an example? MVC is a basic topic which is applicable to a large class of applications; so we could do MVC and then explain Rails a bit as an example. Thoughts?

I mean MVC is used not just in webapps but also in [certain desktop applications](https://developer.mozilla.org/en-US/docs/Mozilla/Displaying_Place_information_using_views.) So this would be interesting and useful.